### ISSUE TITLE
Set $NOTIFY_SOCKET for ExecStop= processes, and process notify messages from them

### ISSUE THREAD

**Issue by joukewitteveen** [systemd/systemd#4212]
### Submission type
- [X] Bug report
- [ ] Request for enhancement (RFE)

_NOTE: Do not submit anything other than bug reports or RFEs via the issue tracker!_
### systemd version the issue has been seen with

> 231

_NOTE: Do not submit bug reports about anything but the two most recently released systemd versions upstream!_
### Used distribution

> Arch Linux
### In case of bug report: Expected behaviour you didn't see

> Updating the unit status string through `systemd-notify --pid --status="this service is no longer running so the old status should be invalidated"` should update the unit status string.
### In case of bug report: Unexpected behaviour you saw

> No update could be performed as $NOTIFY_SOCKET was not set for the 'ExecStop' process. This happens when the service type is 'notify' and 'NotifyAccess' is not set (implicitly set to 'main'). I feel that the ExecStop process should behave as the main process.
### In case of bug report: Steps to reproduce the problem

> Try to update the process status from a program run via 'ExecStop'.

**Comment by joukewitteveen** [comment]
I think the offending piece of code is the presence of `EXEC_IS_CONTROL` in `src/core/service.c:service_enter_stop`. Is it very unreasonable to expect the `ExecStop` process to behave as the main process?

**Comment by davidstrauss** [comment]
I'm not sure I understand the use case. An `ExecStop=` should perform actions to shut down the daemon, which should cause the daemon tell the notify socket that it's stopping.

**Comment by joukewitteveen** [comment]
For a `RemainAfterExit=` daemon, all stopping functionality takes place in the `ExecStop=` process. Consider the following:

```
[Service]
Type=notify
RemainAfterExit=yes
ExecStart=systemd-notify --pid --status="Going strong" --ready
ExecStop=systemd-notify --pid --status=""
```

This does not work as desired because the `ExecStop=` process is not allowed to notify. Of course, this can be solved by opening up `NotifyAccess=`, but this feels like overkill. Maybe make `ExecStop=` the main process if the `ExecStart=` process has already exited? Or include more granularity (`NotifyAccess=toplevel` for access only to processes started by systemd itself)?

**Comment by poettering** [comment]
Hmm, I figure we could also set the env var for ExecStop=, at least depending on some setting in NotifyAccess= as proposed.

**Comment by evverx** [comment]
Closed by #4745
